src/mapgen_v7.cpp

```diff
@@ -483,25 +480,34 @@ void MapgenV7::carveRivers() {
 
 int MapgenV7::generateTerrain()
 {
-	int ymax = generateBaseTerrain();
+	int stone_surface_max_y = -MAP_GENERATION_LIMIT;
+	bool generate_mountain = false;
+
+	generateBaseTerrain(&stone_surface_max_y, &generate_mountain);
+
+	if ((spflags & MGV7_MOUNTAINS) && generate_mountain) {
```

Comment:
Hrmmm, now that generateTerrain() doesn't just generateTerrain() but also calculateNoise()s, I wonder if these should be named something less misleading.